### DISCUSSION TITLE
Expose method to generate slug

### DISCUSSION THREAD

**Discussion by marceloverdijk** [withastro/roadmap#1056] [Proposal]
Internally Astro definitely has a slug generator.

It would be nice to expose this so users can generate a slug from a string text (eg a blog category like “Health Improvements” becomes “health-improvements”).

I can easily create a custom slugify function or use an external dependency, but as Astro already does this internally I would prefer to use that for consistent slugs.

https://discord.com/channels/830184174198718474/1307294084045606953

**Comment by ematipico** [comment]
Can you please use the template we provide? 

Regarding the proposal, you might want to add some code to better frame the suggestion, it's not very clear how/when someone would use this function

**Reply by marceloverdijk** [reply]
Hi @ematipico ,

My use case is I have content collection that defines blog categories like:

```
- name: Health
- name: Technology
- name: Personal Development
- name: Food and Recipes
- name: Travel
- name: Arts and Culture
```

For each I want to generate a page with `getStaticPaths` and I want to use a slug derived from the `name` property.
